client/app/telecom/telephony/line/phone/codec/telecom-telephony-line-phone-codec.controller.js

```diff
@@ -29,36 +36,34 @@ angular.module("managerApp").controller("TelecomTelephonyLinePhoneCodecCtrl", fu
         });
     };
 
+    self.hasChanged = function () {
+        return self.model.codecs && self.codecs.value !== self.model.codecs.value;
+    };
+
     /* -----  End of HELPERS  ------*/
 
     /*= ==============================
     =            ACTIONS            =
     ===============================*/
 
-    self.editCodec = function () {
-        var modal = $uibModal.open({
-            animation: true,
-            templateUrl: "app/telecom/telephony/line/phone/codec/edit/telecom-telephony-line-phone-codec-edit.html",
-            controller: "TelecomTelephonyLinePhoneCodecEditCtrl",
-            controllerAs: "CodecEditCtrl",
-            resolve: {
-                lineItem: function () {
-                    return self.line;
-                }
-            }
-        });
+    self.saveNewCodec = function () {
+
+        self.loading.save = true;
+
+        return self.line.saveOption("codecs", self.model.auto ? self.model.codecs.value + "_a" : self.model.codecs.value).then(function () {
+            self.saved = true;
+            Toast.success([$translate.instant("telephony_line_phone_codec_edit_codec_save_success")]);
+            refreshCodecs();
 
-        modal.result.then(refreshCodecs, function (error) {
-            if (error) {
-                if (error.type === "API" && !error.init) {
-                    Toast.error([$translate.instant("telephony_line_phone_codec_edit_codec_save_error"), (error.data && error.data.message) || ""].join(" "));
-                } else if (error.type === "API" && error.init) {
-                    Toast.error([$translate.instant("telephony_line_phone_codec_edit_codec_load_error"), (error.data && error.data.message) || ""].join(" "));
-                }
+        }, function (error) {
+            if (error.type === "API" && !error.init) {
```

Comment:
`error.type === "API"` will never occur because we have this from modal controller:

```js
...
return self.cancel(angular.extend(error, {		
    type: "API"		
}));
```

Reply:
Sure ? I delete modal controller because it has become useless with modal remove